### PR TITLE
Update license gathering script for forked code

### LINKED ISSUE #103551: third_party/forked/golang/LICENSE  is not in LICENSES
we have one in k/k and one in k/utils, both need to be in LICENSES

/assign

### PR DESCRIPTION
**Pull request by dims**
in k/k and k/utils we have some LICENSE files for code we forked. let's gather them up as well.

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/103551

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: hack/update-vendor-licenses.sh [hunk 1]
```diff
@@ -259,7 +259,14 @@ __EOF__
   mv "${TMP_LICENSE_FILE}" "${dest_dir}/LICENSE"
 done
 
+# copy licenses for forked code from vendor and third_party directories
+(cd "${KUBE_ROOT}" && \
+  find vendor third_party -iname 'licen[sc]e*' -o -iname 'notice*' -o -iname 'copying*' | \
+  grep -E 'third_party|forked' | \
+  xargs tar -czf - | tar -C "${TMP_LICENSES_DIR}" -xzf -)
```
Comment (ehashman) [inline hack/update-vendor-licenses.sh:266]: :raised_eyebrow:

**Review by ehashman — COMMENTED** [review]
/lgtm
!

**Comment by pacoxu** [comment]
/retest
🤷‍♂️

**Comment by nikhita** [comment]
/retest

:snowflake:

**Pull request merged**